thavalon-server/src/game/builder.rs

```diff
@@ -30,14 +31,17 @@ impl GameBuilder {
     }
 
     /// Start the game. This consumes `self` because no new players can be added once the game starts.
-    /// The returned `task::JoinHandle` will complete once the game has ended.
-    pub fn start(self) -> task::JoinHandle<()> {
+    /// The returned [`task::JoinHandle`] will complete once the game has ended. The [`Snapshots`] may be
+    /// used to track per-player snapshots of the game state.
+    pub fn start(self) -> (Snapshots, task::JoinHandle<()>) {
         let game = Game::roll(self.players);
-        let mut interactions = self.interactions;
-        task::spawn(async move {
+        let mut interactions = SnapshotInteractions::new(self.interactions);
```

Comment:
Does ChannelInteractions implement Copy, or are you allowed to move self.interactions out of the struct, even though interactions isn't an `Option`, because you own it now?

Reply:
The second - since this takes ownership of `self`, it's allowed to move fields out of the struct as long as the struct isn't used afterwards. So after this line, we can't call any more methods on `GameBuilder` for example